### PR TITLE
Add category optionsmenu #852

### PR DESCRIPTION
**Pull request by Glowbal**
* Adds a category selection to the options menu, both client and config export variants.
* Two of the larger modules with settings received categories with this PR. All others still require adding them.
* All categories are localized for clients. 

Adding a new category is done through adding the following attribute to a settings entry:
```c++
category = "STR_ACE_Module_Category";
```
Obviously replace module by whatever module the setting belongs to and add the string to the stringtable entry with necessary translations. No other requirements.

### CODE REVIEW

**Comment by jonpas** [comment]
> `category = "STR_ACE_Module_Category";`

You mean `category = "$STR_ACE_Module_Category";`

**Comment by MikeMatrix** [comment]
Uh interesting. Definitely have to take a look at this later.
Something we desperately needed.

**Comment by Glowbal** [comment]
No, I don't. Using $ would mean the clients receive the localized string from the server.

**Comment by jonpas** [comment]
Ah I see, my bad!

**Comment by Glowbal** [comment]
You made me double check the macro I used though -  turns out I did use the wrong one. Fixed now.

**Comment by jonpas** [comment]
Hehe, glad I helped :banana:

File: addons/interact_menu/stringtable.xml [hunk 1]
```diff
@@ -267,5 +267,8 @@
             <Czech>Přidá možnost interakce pro otevření dvěří a umistňovat žebříky na budovy. (Poznámka: Použití této možnosti snižuje výkon při otevírání pomocí interakčního menu, zejména ve velkých městech.) </Czech>
             <Spanish>Añade las acciones de interacción para la apertura de puertas y montaje de escaleras en los edificios. (Nota: Hay un coste de rendimiento al abrir el menú de interacción, especialmente en las ciudades)</Spanish>
         </Key>
+        <Key ID="STR_ACE_Interact_Menu_Category_InteractionMenu">
+            <English>Interaction Menu</English>
+        </Key>
```
Comment (gienkov) [inline addons/interact_menu/stringtable.xml]: ```<Polish>Menu interakcji</Polish>```

File: addons/optionsmenu/stringtable.xml [hunk 2]
```diff
@@ -367,5 +367,8 @@
             <Polish>Pokazuj wiadomości ACE w menu głównym</Polish>
             <Czech>Zobrazit novinky v hlavním menu</Czech>
         </Key>
+        <Key ID="STR_ACE_OptionsMenu_category_all">
+            <English>All Categories</English>
+        </Key>
```
Comment (gienkov) [inline addons/optionsmenu/stringtable.xml]: ```<Polish>Wszystkie kategorie</Polish>```

**Comment by SzwedzikPL** [comment]
It would be great to see options menu in CBA

**Pull request merged**